iroh-net/src/hp/derp/map.rs

```diff
@@ -57,6 +57,18 @@ impl DerpMap {
 
         dm
     }
+
+    /// Returns the [`DerpNode`] by name.
+    pub fn named_node(&self, node_name: &str) -> Option<&DerpNode> {
+        for r in self.regions.values() {
+            for n in &r.nodes {
+                if n.name == node_name {
+                    return Some(n);
+                }
+            }
```

Comment:
could be written as an iterator 

```rust
self.regions
  .values()
  .flat_map(|r| r.nodes.iter())
  .find(|n| n.name == node_name)
```

no strong preference either direction, just wanted to mention it.